src/OpenColorIO/ops/Gamma/GammaOpData.cpp

```diff
@@ -0,0 +1,1248 @@
+inline bool IsMonCurveIdentity(const GammaOpData::Params & p)
+{
+    return (p[0] == IdentityScale && p[1] == IdentityOffset);
+}
+
+std::string GetParametersString(const GammaOpData::Params & params)
+{
+    std::ostringstream oss;
+    oss.precision(FLOAT_DECIMALS);
+    oss << params[0];
+    for(size_t idx=1; idx<params.size(); ++idx)
+    {
+        oss << ", " << params[idx];
+    }
+    return oss.str();
+}
+
+}; // anon
+
+
+const char * GammaOpData::convertStyleToString(Style style)
+{
+    switch(style)
+    {
+        case BASIC_FWD:
+            return "basicFwd";
+            break;
+        case BASIC_REV:
+            return "basicRev";
+            break;
+        case MONCURVE_FWD:
+            return "moncurveFwd";
+            break;
+        case MONCURVE_REV:
+            return "moncurveRev";
+            break;
+    }
+
+    std::stringstream ss("Unknown Gamma style: ");
+    ss << style;
+
+    throw Exception(ss.str().c_str());
+
+    return "basicFwd";
+}
+
+GammaOpData::GammaOpData()
+    :   OpData(BIT_DEPTH_F32, BIT_DEPTH_F32)
+    ,   m_style(BASIC_FWD)
+    ,   m_redParams(getIdentityParameters(m_style))
+    ,   m_greenParams(getIdentityParameters(m_style))
+    ,   m_blueParams(getIdentityParameters(m_style))
+    ,   m_alphaParams(getIdentityParameters(m_style))
+{
+}
+
+GammaOpData::GammaOpData(BitDepth inBitDepth,
+                         BitDepth outBitDepth,
+                         const std::string & id,
+                         const OpData::Descriptions & desc,
+                         const Style & style,
+                         const Params & redParams,
+                         const Params & greenParams,
+                         const Params & blueParams,
+                         const Params & alphaParams)
+    :   OpData(inBitDepth, outBitDepth, id, desc)
+    ,   m_style(style)
+    ,   m_redParams(redParams)
+    ,   m_greenParams(greenParams)
+    ,   m_blueParams(blueParams)
+    ,   m_alphaParams(alphaParams)
+{
+}
+
+GammaOpData::~GammaOpData()
+{
+}
+
+GammaOpDataRcPtr GammaOpData::inverse() const
+{
+    Style invStyle = BASIC_FWD;
+    switch(getStyle())
+    {
+      case BASIC_FWD:     invStyle = BASIC_REV;     break;
+      case BASIC_REV:     invStyle = BASIC_FWD;     break;
+      case MONCURVE_FWD:  invStyle = MONCURVE_REV;  break;
+      case MONCURVE_REV:  invStyle = MONCURVE_FWD;  break;
+    }
+
+    return std::make_shared<GammaOpData>(getOutputBitDepth(), getInputBitDepth(),
+                                         "", Descriptions(), invStyle,
+                                         getRedParams(), getGreenParams(),
+                                         getBlueParams(), getAlphaParams() );
+}
+
+bool GammaOpData::isInverse(const GammaOpData & B) const
+{
+    const Style styleA = getStyle();
+    const Style styleB = B.getStyle();
+
+    // Note: It's possible that someone could create something where they

```

Comment:
Perhaps this edge case should trigger a warning, so its not quietly unhandled.

Reply:
When reading from config. files or CTF file, the validation traps the constraints (refer to GammaOpData::validateParameters() ).

Reply:
Michael, does Patrick's reply resolve your concern?

Reply:
Yes, thanks.